client/src/app/components/identity/IdentityStardustResolver.tsx

```diff
@@ -0,0 +1,256 @@
+import React, { Fragment, ReactNode } from "react";
+import { RouteComponentProps } from "react-router-dom";
+import welcomeMessage from "../../../assets/modals/identity-resolver/welcome.json";
+import { ServiceFactory } from "../../../factories/serviceFactory";
+import { IdentityDiffStorageService } from "../../../services/identityDiffStorageService";
+import { IdentityService } from "../../../services/identityService";
+import AsyncComponent from "../AsyncComponent";
+import "./IdentityStardustResolver.scss";
+import CopyButton from "../CopyButton";
+import JsonViewer from "../JsonViewer";
+import Modal from "../Modal";
+import Spinner from "../Spinner";
+import { IdentityStardustResolverProps } from "./IdentityStardustResolverProps";
+import { IdentityStardustResolverState } from "./IdentityStardustResolverState";
+
+class IdentityStardustResolver extends AsyncComponent<
+    RouteComponentProps<IdentityStardustResolverProps> & { protocolVersion: string },
+    IdentityStardustResolverState
+> {
+    /**
+     * Timer to check to state update.
+     */
+    private _timerId?: NodeJS.Timer;
+
+    /**
+     * placeholder when messageId is not available.
+     */
+    private readonly EMPTY_MESSAGE_ID = "0".repeat(64);
+
+    constructor(props: RouteComponentProps<IdentityStardustResolverProps> & { protocolVersion: string }) {
+        super(props);
+
+
+        this.state = {
+            did: props.match.params.did,
+            aliasId: getAliasId(props.match.params.did ?? ""),
+            error: false,
+            errorMessage: ""
+        };
+    }
+
+    public async componentDidMount(): Promise<void> {
+        super.componentDidMount();
+
+        if (!this.state.did) {
+            return;
+        }
+
+        const res = await ServiceFactory.get<IdentityService>("identity").resolveIdentityStardust(
+            this.state.did,
+            this.props.match.params.network
+        );
+
+        if (res.error) {
+            this.setState({
+                error: true,
```

Comment:
Consider having just one optional `error` state field which carries the error message. I don't see the need for two fields.